### DISCUSSION TITLE
Today Electric

### DISCUSSION THREAD

**Discussion by peterodro** [jasonacox/tinytuya#239] [Q&A]
How to get "today electric kWh"?
DPS '17' is "Add Electricity' - what is meaning of this?
when reading DPS 17 it get "100" but in apk for today i see 0.34kWh

thanks

**Comment by jasonacox** [comment]
Hi @peterodro !  All of the Tuya devices I have only report instantaneous wattage. I sample the wattage every 5m and use that to compute the area under the curve to get kWh for a given time duration.  I think the issue with polling the device for kWh is that it doesn't know what time duration you want (start to end) and would need to do something similar to what I'm doing to compute kWh over time.  I suppose some may have a "lifetime" kWh memory but I haven't found that yet.

**Reply by peterodro** [reply]
Hi, thanks for answer.
in meantime i did same.
im read actual power every 30s then I calcuclae used eneregy as P  * t (t is 30/3600 hrs)  and add it to previous value.
and reset at midnight...
not so precise but ok
i have also using tapo plug P100, there is kwh readble. But it is not tuya...
